In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Projet Generali Challenge

Lecture du fichier

In [3]:
data = pd.read_csv('X.csv', delimiter=',')
data.head()

Unnamed: 0  Identifiant  ft_2_categ         EXPO  ft_4_categ ft_5_categ  \
0           0        18702        2014            1           0          V   
1           1         3877        2014            1           0          V   
2           2         4942        2013            1           1          V   
3           3        13428        2013  0,246575342           0          N   
4           4        17137        2015            1           0          V   

  ft_6_categ ft_7_categ ft_8_categ ft_9_categ  ... ft_16_categ ft_17_categ  \
0          N          1          O          1  ...   IDENTIQUE           V   
1          V          V          V          V  ...           V           N   
2          V          V          V          V  ...           V           N   
3          V          V          V          V  ...           V           N   
4          N          2          O          1  ...   IDENTIQUE           V   

  ft_18_categ ft_19_categ superficief ft_21_categ ft_22_categ ft_23_categ  \
0        base           2      1351.0           4      2012.0         0.0   
1        base           2      1972.0           2      1980.0         0.0   
2        base           2      1630.0           4         NaN         0.0   
3        base           2       532.0           3         NaN         0.0   
4        base           2      1050.0           2      1972.0         0.0   

  ft_24_categ  Insee  
0           2  65440  
1           .  14341  
2           .  75109  
3           .  92004  
4           4  59340  

[5 rows x 26 columns]

In [4]:
df = data.copy()

## Dimensions du dataset

In [5]:
df.shape

(10229, 26)

Nous avons 10229 lignes, une colonne binaire à prédire et 25 variables explicatives.

## Types et valeurs manquantes par variable

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10229 entries, 0 to 10228
Data columns (total 26 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Unnamed: 0   10229 non-null  int64  
 1   Identifiant  10229 non-null  int64  
 2   ft_2_categ   10229 non-null  int64  
 3   EXPO         10229 non-null  object 
 4   ft_4_categ   10229 non-null  int64  
 5   ft_5_categ   10229 non-null  object 
 6   ft_6_categ   10229 non-null  object 
 7   ft_7_categ   10229 non-null  object 
 8   ft_8_categ   10229 non-null  object 
 9   ft_9_categ   10229 non-null  object 
 10  ft_10_categ  10229 non-null  object 
 11  ft_11_categ  10229 non-null  object 
 12  ft_12_categ  10229 non-null  object 
 13  ft_13_categ  10229 non-null  object 
 14  ft_14_categ  10229 non-null  object 
 15  ft_15_categ  10229 non-null  object 
 16  ft_16_categ  10229 non-null  object 
 17  ft_17_categ  10229 non-null  object 
 18  ft_18_categ  10229 non-null  object 
 19  ft_1

In [7]:
df.isnull().sum()

Unnamed: 0        0
Identifiant       0
ft_2_categ        0
EXPO              0
ft_4_categ        0
ft_5_categ        0
ft_6_categ        0
ft_7_categ        0
ft_8_categ        0
ft_9_categ        0
ft_10_categ       0
ft_11_categ       0
ft_12_categ       0
ft_13_categ       0
ft_14_categ       0
ft_15_categ       0
ft_16_categ       0
ft_17_categ       0
ft_18_categ       0
ft_19_categ       0
superficief     119
ft_21_categ       0
ft_22_categ    1236
ft_23_categ       0
ft_24_categ       0
Insee           115
dtype: int64

Observations : Nous avons 119 valeurs nulles dans la colonne superficief, 1236 dans la colonne ft_22_categ ainsi que 115 dans Insee.

On pourrait remplacer les valeurs manquantes par une imputation avancée (entraîner le modèle pour prédire les données manquantes)

In [8]:
df[df['superficief'].isna()]

Unnamed: 0  Identifiant  ft_2_categ         EXPO  ft_4_categ  \
15             15        14454        2013            1           1   
33             33        17151        2012            1           0   
37             37        11721        2012            1           0   
45             45        15994        2012            1           1   
136           136         5078        2013            1           1   
...           ...          ...         ...          ...         ...   
9711         9711        15067        2012            1           0   
9729         9729        17650        2012            1           1   
10007       10007        17249        2012            1           0   
10071       10071           48        2014  0,030136986           0   
10124       10124        15812        2012            1           0   

      ft_5_categ ft_6_categ ft_7_categ ft_8_categ ft_9_categ  ... ft_16_categ  \
15             V          V          V          V          V  ...           V   
33             V          V          V          V          V  ...           V   
37             V          V          V          V          V  ...           V   
45             V          V          V          V          V  ...           V   
136            V          V          V          V          V  ...           V   
...          ...        ...        ...        ...        ...  ...         ...   
9711           V          V          V          V          V  ...           V   
9729           V          V          V          V          V  ...           V   
10007          V          V          V          V          V  ...           V   
10071          V          V          V          V          V  ...           V   
10124          V          V          V          V          V  ...           V   

      ft_17_categ ft_18_categ ft_19_categ superficief ft_21_categ ft_22_categ  \
15              V           V           2         NaN           2      2011.0   
33              V           V           2         NaN           2      1920.0   
37              V           V           2         NaN           2      1993.0   
45              V           V           2         NaN           4      1987.0   
136             V           V           2         NaN           2      1950.0   
...           ...         ...         ...         ...         ...         ...   
9711            V           V           2         NaN           3      1960.0   
9729            V           V           2         NaN           3      1960.0   
10007           V           V           2         NaN           2      1960.0   
10071           V           V           2         NaN           1      1970.0   
10124           V           V           2         NaN           1      2009.0   

      ft_23_categ ft_24_categ  Insee  
15              V           .    NaN  
33              V           .    NaN  
37              V           .    NaN  
45              V           .    NaN  
136             V           .    NaN  
...           ...         ...    ...  
9711            V           .    NaN  
9729            V           .    NaN  
10007           V           .    NaN  
10071           V           .    NaN  
10124           V           .    NaN  

[119 rows x 26 columns]

Observations : 
- 